lib/manager.js

```diff
@@ -286,25 +284,24 @@ internals.Manager.prototype.registerHelper = function (name, helper) {
 };
 
 
-internals.Manager.prototype.render = function (filename, context, options, callback) {
-
-    if (!callback) {
-        return internals._wrapMethod(this, this.render, [filename, context, options]);
-    }
-
-    this._prepare(filename, options, (err, compiled) => {
+internals.Manager.prototype.render = function (filename, context, options, request) {
 
-        if (err) {
-            return callback(err);
-        }
+    return new Promise((resolve, reject) => {
 
-        this._render(compiled, context, null, (err, rendered) => {
+        this._prepare(filename, options, (err, compiled) => {
 
             if (err) {
-                return callback(err);
+                return reject(err);
             }
 
-            return callback(null, rendered, compiled.settings);
```

Comment:
As for compiled.settings, it wasn't used anywhere in tests and promises can only resolve to 1 value. Looking at `_wrapMethod`, I think this was never being passed anyway since only `err` and `result` were being pulled from the callback https://github.com/hapijs/vision/pull/140/files#diff-b840698050e033004db3ac04392b9ea1L633.